tests/integration/cqlengine/__init__.py

```diff
@@ -44,3 +52,57 @@ def setup_connection(keyspace_name):
                      consistency=ConsistencyLevel.ONE,
                      protocol_version=PROTOCOL_VERSION,
                      default_keyspace=keyspace_name)
+
+
+class StatementCounter(object):
+    """
+    Simple python object used to hold a count of the number of times
+    the wrapped method has been invoked
+    """
+    def __init__(self, patched_func):
+        self.func = patched_func
+        self.counter = 0
+
+    def wrapped_execute(self, *args, **kwargs):
+        self.counter += 1
+        return self.func(*args, **kwargs)
+
+    def get_counter(self):
+        return self.counter
+
+
+def execute_count(expected):
+    """
+    A decorator used wrap cassandra.cqlengine.connection.execute. It counts the number of times this method is invoked
+    then compares it to the number expected. If they don't match it throws an assertion error.
+    This function can be disabled by running the test harness with the env variable CQL_SKIP_EXECUTE=1 set
+    """
+    def innerCounter(fn):
+        def wrapped_function(*args, **kwargs):
+            # Create a counter monkey patch into cassandra.cqlengine.connection.execute
+            count = StatementCounter(cassandra.cqlengine.connection.execute)
+            original_function = cassandra.cqlengine.connection.execute
+            # Monkey patch in our StatementCounter wrapper
+            cassandra.cqlengine.connection.execute = count.wrapped_execute
+            # Invoked the underlying unit test
+            to_return = fn(*args, **kwargs)
+            # Get the count from our monkey patched counter
+            count.get_counter()
+            # DeMonkey Patch our code
+            cassandra.cqlengine.connection.execute = original_function
+            # Check to see if the count is what you expect
+            tc = unittest.TestCase("__init__")
```

Comment:
This creates a new test instance that doesn't seem to fail the one we're wrapping. Could we pass in a reference to the existing test and use that to assert?